cpython-unix/build-zstd.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+set -ex
+
+ROOT=`pwd`
+
+export PATH=${TOOLS_PATH}/${TOOLCHAIN}/bin:${TOOLS_PATH}/host/bin:$PATH
+export PREFIX="/tools/deps"
+
+tar -xf zstd-${ZSTD_VERSION}.tar.gz
+
+pushd cpython-source-deps-zstd-${ZSTD_VERSION}/lib
+
+if [ "${CC}" = "musl-clang" ]; then
+    # In order to build the library with SSE2, BMI, and AVX2 intrinstics, we need musl-clang to find
```

Comment:
This is modeled off of 

https://github.com/astral-sh/python-build-standalone/blob/13731b0e41e6846e0cc12dd0deb942d63d801bf6/cpython-unix/build-cpython.sh#L410-L422

Reply:
Annoyingly, I didn't realize this was the problem until I was pretty far in. We might want to generalize this at some point.